contentcuration/contentcuration/view/internal_views.py

```diff
@@ -0,0 +1,258 @@
+import json
+import logging
+import os
+import re
+import shutil
+from django.http import Http404, HttpResponse, HttpResponseBadRequest, HttpResponseRedirect
+from django.views.decorators.csrf import csrf_exempt
+from django.shortcuts import render, get_object_or_404, redirect, render_to_response
+from django.contrib.auth.decorators import login_required
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.context_processors import csrf
+from django.db.models import Q
+from django.template.loader import render_to_string
+from contentcuration.models import Exercise, AssessmentItem, Channel, License, FileFormat, File, FormatPreset, ContentKind, ContentNode, ContentTag, Invitation, generate_file_on_disk_name
+from le_utils.constants import content_kinds
+from django.db.models.functions import Concat
+from django.core.files import File as DjFile
+from django.db.models import Q, Value
+from django.db import transaction
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication, TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+
+@api_view(['POST'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAuthenticated,))
+def authenticate_user_internal(request):
+    logging.debug("Logging in user")
+    return HttpResponse(json.dumps({'success': True, 'token': unicode(request.auth), 'username':unicode(request.user)}))
+
+@api_view(['POST'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAuthenticated,))
+def file_diff(request):
+    logging.debug("Entering the file_diff endpoint")
+    if request.method != 'POST':
+        return HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        data = json.loads(request.body)
+        in_db_list = File.objects.annotate(filename=Concat('checksum', Value('.'),  'file_format')).filter(filename__in=data).values_list('filename', flat=True)
+        to_return = []
+        for f in list(set(data) - set(in_db_list)):
```

Comment:
Would it make sense to create a file object here without a reference to a contentnode or assessmentitem? Right now, file objects are created only when the channel is being created. If the Ricecooker doesn't complete the create_channel method, it ends up counting the other files as not being uploaded and re-uploads them again even though they are already uploaded